Hometask2

AlexanderKiselyov opened the pull request.

AlexanderKiselyov committed:
Create App.config

AlexanderKiselyov committed:
Add files via upload

AlexanderKiselyov committed:
Create Assembly.info

AlexanderKiselyov committed:
Create LocalNetworkTest.cs

AlexanderKiselyov committed:
Add files via upload

AlexanderKiselyov committed:
Create Assembly,info

AlexanderKiselyov committed:
Rename Assembly,info to AssemblyInfo.cs

AlexanderKiselyov committed:
Rename Assembly.info to AssemblyInfo.cs

AlexanderKiselyov committed:
Add files via upload

yurii-litvinov commented on Homework 2/Computer.cs:
```diff
@@ -0,0 +1,42 @@
+﻿namespace Homework_2
+{
+	public class Computer
```
Нужен комментарий. Ко всему, что public, нужен комментарий

yurii-litvinov commented on Homework 2/Computer.cs:
```diff
@@ -0,0 +1,42 @@
+		/// <summary>
+		/// Class for each local computer.
+		/// </summary>
+		public class LocalComputer
```
А зачем он вложенный?

yurii-litvinov commented on Homework 2/LocalNetwork.cs:
```diff
@@ -0,0 +1,90 @@
+						{
+							if (i != j && linkedComputers[i, j] && !allComputers[j].Infected && !infectedComputers.Contains(allComputers[j]))
+							{
+								random = new Random();
```
Random инициализируется значением текущего времени в милисекундах, поэтому все Random-ы, созданные в одну миллисекунду, будут всегда возвращать одно и то же значение

yurii-litvinov commented on LocalNetworkTest/LocalNetworkTest.cs:
```diff
@@ -0,0 +1,54 @@
+			};
+			var localNetwork = new Network.LocalNetwork(localComputers, linkMatrix);
+			localNetwork.StepOfVirusInfection();
+			Assert.AreEqual(false, localNetwork.IfAllComputersAreInfected());
```
Это же Assert.IsFalse

yurii-litvinov commented on LocalNetworkTest/LocalNetworkTest.csproj:
```diff
@@ -0,0 +1,89 @@
+    </Otherwise>
+  </Choose>
+  <ItemGroup>
+    <Compile Include="LocalNetworkTests.cs" />
```
А файл называется LocalNetworkTest.cs, кто-то не проверяет перед отправкой? :)

yurii-litvinov reviewed (changes requested):
Тесты не проходят, поправьте, пожалуйста

AlexanderKiselyov committed:
Delete Computer.cs

AlexanderKiselyov committed:
Add files via upload

Убраны вложенные файлы, добавлены комментарии.

AlexanderKiselyov committed:
Delete LocalNetwork.cs

AlexanderKiselyov committed:
Add files via upload

Убраны вложенные классы, добавлены комментарии, исправлен вызов функции random.

AlexanderKiselyov committed:
Rename LocalNetworkTest.cs to LocalNetworkTests.cs

Исправлено название класса.

AlexanderKiselyov committed:
Delete LocalNetworkTests.cs

AlexanderKiselyov committed:
Add files via upload

Исправлено название класса.

yurii-litvinov commented on Homework 2/LocalNetwork.cs:
```diff
@@ -0,0 +1,86 @@
+							if (random.Next(0, 100) < allComputers[j].InfectionProbability())
+							{
+								allComputers[j].Infected = true;
+								infectedComputers.Add(allComputers[j]);
```
Я бы это доверил самому компьютеру, он знает свою ОС и у него есть поле, отвечающее за то, заражён ли он
